comp/dogstatsd/server/parse.go

```diff
@@ -239,7 +239,10 @@ func (p *parser) parseFloat64List(rawFloats []byte) ([]float64, error) {
 
 // extractContainerID parses the value of the container ID field.
 func (p *parser) extractContainerID(rawContainerIDField []byte) []byte {
-	return rawContainerIDField[len(containerIDFieldPrefix):]
+	containerIdRef := rawContainerIDField[len(containerIDFieldPrefix):]
+	containerId := make([]byte, len(containerIdRef))
+	copy(containerId, containerIdRef)
+	return containerId
```

Comment:
`lading` currently generates this field for metrics but not events or service checks.

Reply:
This does look like a potential issue though; container IDs will keep the whole packet buffer from being garbage collected.